api/src/integrations/core/datasource.js

```diff
@@ -250,42 +250,65 @@ class CoreDataSource {
   }
 
   /**
-   * Gets valid permit for a Core mine. A valid permit must meet the following criteria:
-   *  - Must not be exploratory
+   * Gets all valid permits for a Core mine. A valid permit must meet the following criteria:
    *  - Must not be historical
    * .
    *
    * @param {object} nrptiRecord NRPTI record
    * @returns {object} Valid permit.
    * @memberof CoreDataSource
    */
-  async getMinePermit(nrptiRecord) {
+  async getMinePermits(nrptiRecord) {
     if (!nrptiRecord) {
-      throw Error('getMinePermit - required nrptiRecord is null.');
+      throw Error('getMinePermits - required nrptiRecord is null.');
     }
 
     // Get permits with detailed information.
     const url = getIntegrationUrl(CORE_API_HOST, `/api/mines/${nrptiRecord._sourceRefId}/permits`);
     const { records: permits } = await integrationUtils.getRecords(url, getAuthHeader(this.client_token));
 
-    // First, any mines with 'X' as their second character are considered exploratory. Remove them.
-    // const nonExploratoryPermits = permits.filter(permit => permit.permit_no[1].toLowerCase() !== 'x');
+    return permits.filter(p => this.isValidPermit(p,nrptiRecord));
+  }
+
+  isValidPermit(permit,nrptiRecord){
+    //Mine must not be historical which is indicated by an authorized year of '9999' on the latest amendment.
+    if((permit.permit_amendments.length && !permit.permit_amendments[0].authorization_end_date) 
+      || permit.permit_status_code === 'O'){
```

Comment:
Just minor formatting inconsistencies to the rest of the file with regards to whitespace

Reply:
Good spotting! I did see this when I reviewed the PR too, surprised the linter didn't catch it